File: threedollar-api/src/main/java/com/depromeet/threedollar/api/service/faq/dto/response/FaqResponse.java
package com.depromeet.threedollar.api.service.faq.dto.response;

import com.depromeet.threedollar.domain.domain.faq.Faq;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class FaqResponse {

	private Long id;

	private String question;

	private String answer;

	public static FaqResponse of(Faq faq) {
		return new FaqResponse(faq.getId(), faq.getQuestion(), faq.getAnswer());
	}

}

File: threedollar-api/src/main/java/com/depromeet/threedollar/api/service/faq/dto/response/FaqResponse.kt
package com.depromeet.threedollar.api.service.faq.dto.response

import com.depromeet.threedollar.api.common.dto.AuditingTimeResponse
import com.depromeet.threedollar.domain.domain.faq.Faq
import com.depromeet.threedollar.domain.domain.faq.FaqCategory

data class FaqResponse(
    val faqId: Long,
    val question: String,
    val answer: String,
    val category: FaqCategory
) : AuditingTimeResponse() {

    companion object {
        fun of(faq: Faq): FaqResponse {
            val response = FaqResponse(faq.id, faq.question, faq.answer, faq.category)
            response.setBaseTime(faq)
            return response
        }
    }

}
